provider/adapters/src/fallback/mod.rs

```diff
@@ -397,6 +398,66 @@ impl<'a> LocaleFallbackerWithConfig<'a> {
             },
         }
     }
+
+    /// Sorts the locales from an iterator into a list of levels based on how far they are
+    /// away from the root locale.
+    ///
+    /// Note: Only those locales that were in the input iterator will be present in the output,
+    /// even if one or more of those locales have a parent that is not in the input.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::locale;
+    /// use icu_provider::prelude::*;
+    /// use icu_provider_adapters::fallback::LocaleFallbackConfig;
+    /// use icu_provider_adapters::fallback::LocaleFallbacker;
+    /// use icu_provider::hello_world::HelloWorldV1Marker;
+    ///
+    /// let fallbacker =
+    ///     LocaleFallbacker::try_new_unstable(&icu_testdata::unstable())
+    ///         .expect("data");
+    /// let key_fallbacker = fallbacker.for_key(HelloWorldV1Marker::KEY);
+    ///
+    /// let locales = [
+    ///     locale!("en-GB"),
+    ///     locale!("en-IN"),
+    ///     locale!("en-US"),
+    ///     locale!("en"),
+    ///     locale!("hi-Latn-IN"),
+    ///     locale!("und"),
+    ///     locale!("zh-CN"),
+    ///     locale!("zh-TW"),
+    ///     locale!("zh"),
+    /// ];
+    ///
+    /// let grouped_locales = key_fallbacker.sort_locales_into_groups(locales.iter().map(|l| l.into()));
+    ///
+    /// assert_eq!(grouped_locales, vec![
+    ///     vec![locale!("und").into()],
+    ///     vec![locale!("en").into(), locale!("zh").into()],
+    ///     vec![locale!("en-US").into(), locale!("zh-CN").into()],
+    ///     vec![locale!("en-GB").into(), locale!("en-IN").into(), locale!("zh-TW").into()],
+    ///     vec![], // note: hi-Latn is not in the input, so it is not in the output
+    ///     vec![locale!("hi-Latn-IN").into()]
+    /// ])
+    /// ```
```

Comment:
`question`: why is `en-US` in a different bucket than `en-GB`?

Reply:
because en-GB goes through en-001.

Reply:
ah, thanks. Can you document that?
I'm still a confused as to why `en-IN` goes through `en-001` and `en-US` does not. Or what is the parent of `zh-TW` - it's not in https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-core/supplemental/parentLocales.json .

Reply:
The behavior of en-IN is the module-level example here:

https://icu4x.unicode.org/doc/icu_provider_adapters/fallback/index.html

zh-TW goes through zh-Hant-TW and zh-Hant; that one is not in the docs but it has a unit test here:

https://github.com/unicode-org/icu4x/blob/b7ea6e67c1a7caf36f43ee2fee23a207f5bc4390/provider/adapters/src/fallback/algorithms.rs#L369

Also I'm considering hiding this function from public API (moving it closer to the call site in datagen) because it's fairly specific and everything can be accomplished via existing public APIs